### ISSUE TITLE
Heaps support?

### ISSUE THREAD

**Issue by Gaboose** [polybiusproxy/hxCodec#270]
Are there any plans to add support for the Heaps engine?

I would imagine this being implemented as another folder `hxcodec/heaps` next to `hxcodec/openfl` and `hxcodec/flixel` with a class that extends [h2d.Bitmap](https://heaps.io/api/h2d/Bitmap.html) and reuses the `hxcodec/vlc` package underneath.

If this is desired, or at least you're ok with it, I could help by writing a pull request :)

**Comment by MAJigsaw77** [comment]
> Are there any plans to add support for the Heaps engine?
> 
> I would imagine this being implemented as another folder `hxcodec/heaps` next to `hxcodec/openfl` and `hxcodec/flixel` with a class that extends [h2d.Bitmap](https://heaps.io/api/h2d/Bitmap.html) and reuses the `hxcodec/vlc` package underneath.
> 
> If this is desired, or at least you're ok with it, I could help by writing a pull request :)

This lib is normally designed to work with Haxe/OpenFL but I don't see why not ig

**Comment by Gaboose** [comment]
On further investigation it seems I can't use LibVLC.hx while targeting hashlink, even if I tell it to output c code (`-hl out/main.c`).

```
./hxCodec/hxcodec/vlc/LibVLC.hx:174: characters 55-90 : You cannot access the cpp package while targeting hl (for cpp.RawConstPointer)
./hxCodec/hxcodec/vlc/VLCBitmap.hx:9: characters 8-26 : ... referenced here
./hxCodec/hxcodec/heaps/VideoBitmap.hx:4: characters 8-29 : ... referenced here
src/Main.hx:1: characters 8-33 : ... referenced here
```

And heaps doesn't support the cpp target ([source](https://community.heaps.io/t/solved-error-when-compiling-heaps-to-cpp-windows/448/2)).

So now I'm not sure how to do this anymore, or if it's even possible.

I already made some preparatory work, I moved `openfl` dependencies from `hxcodec/vlc` to `hxcodec/openfl` in this commit: https://github.com/polybiusproxy/hxCodec/commit/c7be9419f08358e0dc990c54023ee24e6e5596a8. But I guess there's no point in making a pull request now.
